components/metrics/http.go

```diff
@@ -26,12 +26,12 @@ func ServeHTTP(addr string, registry *prometheus.Registry) (cancel func()) {
 	})
 	server := http.Server{
 		Addr:    addr,
-		Handler: handler,
+		Handler: router,
 	}
 
 	go func() {
 		err := server.ListenAndServe()
-		if err != nil {
+		if err != http.ErrServerClosed {
```

Comment:
This will panic for a nil err too.

Reply:
server.ListenAndServe is guaranteeing to return non-nil error. Checking for nil might be more explicit though.

Reply:
Ah, indeed, I only checked the error handling, but not what it actually handles.